.github/workflows/kakarot_rpc.yml

```diff
@@ -1,35 +1,109 @@
+---
 name: RPC
 
 on:
   push:
     branches:
-      - "main"
+      - 'main'
+
+env:
+  REGISTRY_IMAGE: ghcr.io/${{ github.repository }}/node
 
 jobs:
-  build_and_push:
+  build:
+    runs-on: ubuntu-latest-32-cores
     strategy:
       fail-fast: false
       matrix:
         platform:
           - linux/amd64
           - linux/arm64
-    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY_IMAGE }}
+          tags: |
+            type=raw,value=latest
+
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
+
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and push
+
+      - name: Build and push by digest
+        id: build
         uses: docker/build-push-action@v4
         with:
-          push: true
-          tags: ghcr.io/kkrt-labs/kakarot-rpc/node:latest
           context: .
+          file: Dockerfile
           platforms: ${{ matrix.platform }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
Caching the docker build https://docs.docker.com/build/cache/backends/gha/#using-dockerbuild-push-action